posthog/tasks/enforce_max_replay_retention_period.py

```diff
@@ -0,0 +1,72 @@
+import logging
+from typing import Optional
+
+from celery import shared_task
+
+from posthog.constants import AvailableFeature
+from posthog.models.organization import ProductFeature
+from posthog.models.team import Team
+
+logger = logging.getLogger(__name__)
+
+
+BATCH_SIZE = 100
+
+
+def _parse_feature_to_retention(retention_feature: Optional[ProductFeature]) -> str | None:
+    if retention_feature is None:
+        return None
+
+    retention_limit: int | None = retention_feature.get("limit")
+    retention_unit: str | None = retention_feature.get("unit")
+
+    if retention_limit is None or retention_unit is None:
+        return None
+
+    match retention_unit.lower():
+        case "day" | "days":
+            highest_retention_entitlement = f"{retention_limit}d"
+        case "month" | "months":
+            if retention_limit < 12:
+                highest_retention_entitlement = f"{retention_limit * 30}d"
+            else:
+                highest_retention_entitlement = f"{retention_limit // 12}y"
+        case "year" | "years":
+            highest_retention_entitlement = f"{retention_limit}y"
+        case _:
+            return None
+
+    if highest_retention_entitlement not in ["30d", "90d", "1y", "5y"]:
+        return None
+
+    return highest_retention_entitlement
+
+
+@shared_task(ignore_result=True)
+def enforce_max_replay_retention_period() -> None:
+    teams_to_update = []
+
+    for team in Team.objects.exclude(session_recording_retention_period="legacy").only(
```

Comment:
i guess you can exclude teams with `30d` set too? 
since they can never be over their max entitlement and will make up the majority of the list